system/assets/js/admin/specific/urlInput/urlInput.js

```diff
@@ -2,17 +2,29 @@
 
 	$.fn.presideUrlInput = function() {
 		return this.each( function() {
-			$( this ).on( "change", function( e ) {
-				var $hidden     = $( this ).closest( ".row" ).next( 'input[type="hidden"]' )
-				  , inputId     = $hidden.attr( "id" )
-				  , $protocol   = $( "#" + inputId + "_protocol" )
+			var $hidden   = $( this )
+			  , inputName = $hidden.attr( "name" )
+			  , inputId   = $hidden.attr( "id" )
+			;
+
+			$( '[name="' + inputName + '_protocol"],#' + inputId + "_domain_path").on( "change", function( e ) {
+				var $protocol   = $( '[name="' + inputName + '_protocol"]' )
 				  , $domainPath = $( "#" + inputId + "_domain_path" )
 				;
 
-				if ( $protocol.val().length > 0 && $domainPath.val().length > 0 ) {
-					var matches = $domainPath.val().match( /^(.*:\/\/)/g );
+				if ( $domainPath.val().length > 0 ) {
```

Comment:
This still doesn't look right?

* If a protocol is detected in domain path then, **always set the protocol** using that value
* If a protocol is _not_ detected in the domain path and protocol field is empty, then set the protocol field to https